viskores/cont/CubicHermiteSpline.h

```diff
@@ -0,0 +1,105 @@
+//============================================================================
+//  The contents of this file are covered by the Viskores license. See
+//  LICENSE.txt for details.
+//
+//  By contributing to this file, all contributors agree to the Developer
+//  Certificate of Origin Version 1.1 (DCO 1.1) as stated in DCO.txt.
+//============================================================================
+
+//============================================================================
+//  Copyright (c) Kitware, Inc.
+//  All rights reserved.
+//  See LICENSE.txt for details.
+//
+//  This software is distributed WITHOUT ANY WARRANTY; without even
+//  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+//  PURPOSE.  See the above copyright notice for more information.
+//============================================================================
+#ifndef viskores_cont_CubicHermiteSpline_h
+#define viskores_cont_CubicHermiteSpline_h
+
+#include <viskores/Range.h>
+#include <viskores/Types.h>
+#include <viskores/cont/ArrayHandle.h>
+#include <viskores/cont/ExecutionObjectBase.h>
+#include <viskores/cont/UnknownArrayHandle.h>
+#include <viskores/cont/viskores_cont_export.h>
+#include <viskores/exec/CubicHermiteSpline.h>
+
+namespace viskores
+{
+namespace cont
+{
+
+class VISKORES_CONT_EXPORT CubicHermiteSpline : public viskores::cont::ExecutionObjectBase
+{
+public:
+  CubicHermiteSpline() = default;
+  virtual ~CubicHermiteSpline() = default;
+
+  VISKORES_CONT void SetData(const viskores::cont::ArrayHandle<viskores::Vec3f>& data)
+  {
+    this->Data = data;
+  }
+  VISKORES_CONT void SetData(const std::vector<viskores::Vec3f>& data,
+                             viskores::CopyFlag copy = viskores::CopyFlag::On)
+  {
+    this->Data.Allocate(0);
+    this->Data = viskores::cont::make_ArrayHandle(data, copy);
+  }
+
+  VISKORES_CONT void SetKnots(const viskores::cont::ArrayHandle<viskores::FloatDefault>& knots)
+  {
+    this->Knots = knots;
```

Comment:
While we are on the subject of clearing the `Knots` and `Tangents`, it might be worthwhile to check if you attempt to set the `Knots` or `Tangents` before the `Data` to give an indication that the set values will be overwritten. It could look like this.

```suggestion
    if (this->Data.GetNumberOfValues() != knots.GetNumberOfValues())
    {
      throw vtkm::cont::ErrorBadValue("Size of CubicHermiteSpline Knots array must be the same size as the Data array. Make sure you set the Data array before the Knots array.");
    }
    this->Knots = knots;
```

Something similar should be added to all `SetKnots` and `SetTangents` methods.